Python/compile.c

```diff
@@ -1850,9 +1850,7 @@ codegen_kwonlydefaults(struct compiler *c, location loc,
                 goto error;
             }
             ADDOP_LOAD_CONST_NEW(c, loc, mangled);
-            if (compiler_visit_expr(c, default_) < 0) {
-                goto error;
-            }
+            RETURN_IF_ERROR(codegen_visit_expr(c, default_));
```

Comment:
This looks like it leaks `keys`, as it skips the `Py_XDECREF(keys);` is skipped.
But appears `keys` is unused in this function. Maybe remove it?